### PR TITLE
Add option to alias with hostname in addition to bosh_id

### PR DESCRIPTION
**Pull request by zippolyte**
### What does this PR do?

Add the option to set the OS hostname as an alias in addition to the bosh_id when running on CloudFoundry, so that it can be used alongside the `friendly_hostname` of the bosh agent (https://github.com/DataDog/datadog-agent-boshrelease/blob/2a3a3fddcbdc58ac2d1f23c94b6bf023825f9633/jobs/dd-agent/templates/config/datadog.yaml.erb#L55-L65).

### Motivation

Use case of PCF installed on vSphere managed VMs, where the vSphere integration sends OS hostnames, so that there is no host duplication, while still being able to use the friendly_hostname to identify the machines (PCF sets the OS hostname of its VMs to a GUID, so they are difficult to identify)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/util/cloudfoundry/cloudfoundry.go [hunk 1]
```diff
@@ -3,7 +3,7 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-2019 Datadog, Inc.
 
-package cloudfoudry
+package cloudfoundry
```
Comment (truthbk) [inline pkg/util/cloudfoundry/cloudfoundry.go]: 🍰

File: pkg/config/config.go [hunk 2]
```diff
@@ -302,6 +302,7 @@ func initConfig(config Config) {
 	// Cloud Foundry
 	config.BindEnvAndSetDefault("cloud_foundry", false)
 	config.BindEnvAndSetDefault("bosh_id", "")
+	config.BindEnvAndSetDefault("cf_os_hostname_aliasing", false)
```
Comment (truthbk) [inline pkg/config/config.go]: should we add some additional documentation to the config template regarding this option and what it entails?

File: pkg/util/cloudfoundry/cloudfoundry_test.go [hunk 3]
```diff
@@ -16,31 +16,51 @@ func TestHostAliasDisable(t *testing.T) {
...
-	assert.Equal(t, "ID_CF", alias)
+	assert.Equal(t, []string{"ID_CF"}, aliases)
+
+	mockConfig.Set("cf_os_hostname_aliasing", true)
+	aliases, err = GetHostAliases()
+	assert.Nil(t, err)
+
+	hostname, _ := os.Hostname()
+	fqdn := util.Fqdn(hostname)
+	if hostname == fqdn {
```
Comment (truthbk) [inline pkg/util/cloudfoundry/cloudfoundry_test.go]: Nit: I see why you're doing this, but this is the sort of test that would not test everything and might hide a dormant bug depending on the machine where it runs. 

What we're currently doing in that `if...else` is very basic, so I'm suggesting we change this as a "best practice" sort of thing. If you could try to mock `util.Fqdn` output to force both cases, that would be best.

**Review by truthbk — COMMENTED** [review]
I don't really see anything wrong with this, just a nit regarding a test.

File: pkg/util/cloudfoundry/cloudfoundry_test.go (hunk 3 above)
Comment (zippolyte) [reply pkg/util/cloudfoundry/cloudfoundry_test.go]: 🙈 
I tried mocking in go once and said to myself, never again. I'll look into it again 👍

File: pkg/config/config.go (hunk 2 above)
Comment (zippolyte) [reply pkg/config/config.go]: I don't think we need to/should. This option will be specific to the bosh agent, and will be configured via the tile, where it is documented (see the two PRs referencing this one).
The `cloud_foundry` and `bosh_id` options which are not documented in the template either.

File: pkg/util/cloudfoundry/cloudfoundry.go [hunk 4]
```diff
@@ -14,6 +14,8 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util"
 )
 
+var utilFqdn = util.Fqdn
```
Comment (truthbk) [inline pkg/util/cloudfoundry/cloudfoundry.go]: We probably want to add a small comment as to why we're doing this :)

nit: wouldn't you prefer a prettier looking name like `getFqdn`?

**Review by truthbk — DISMISSED** [review]
Looks good to me! Just a teeny nit.

File: pkg/util/cloudfoundry/cloudfoundry.go (hunk 4 above)
Comment (zippolyte) [reply pkg/util/cloudfoundry/cloudfoundry.go]: I do :)

**Review by truthbk — APPROVED** [review]
Thanks for adding the relnote! Unfortunately the integration tests are still broken, but it's related to the docker fallout. It should be safe to merge, but up to you to decide when you want to pull the trigger.

File: pkg/util/cloudfoundry/cloudfoundry.go [hunk 5]
```diff
@@ -14,18 +14,37 @@ import (
...
 	if boshID != "" {
-		return boshID, nil
+		aliases = append(aliases, boshID)
 	}
 
 	hostname, _ := os.Hostname()
-	return util.Fqdn(hostname), nil
+	fqdn := getFqdn(hostname)
+
+	if config.Datadog.GetBool("cf_os_hostname_aliasing") {
```
Comment (gzussa) [inline pkg/util/cloudfoundry/cloudfoundry.go]: Why don't we do this all the time?

Comment (zippolyte) [reply pkg/util/cloudfoundry/cloudfoundry.go]: I don't want to potentially break current configs of users

File: releasenotes/notes/pcf_alias-0d227fd4f5d7af93.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# Each section from every releasenote are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
```
Comment (olivielpeau) [inline releasenotes/notes/pcf_alias-0d227fd4f5d7af93.yaml]: `enhancements`

File: releasenotes/notes/pcf_alias-0d227fd4f5d7af93.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  - |
+    Add option ``cf_os_hostname_alias`` to send the OS hostname as an alias when using the BOSH agent on Cloud Foundry
```
Comment (olivielpeau) [inline releasenotes/notes/pcf_alias-0d227fd4f5d7af93.yaml]: ``cf_os_hostname_aliasing``

**Pull request merged**